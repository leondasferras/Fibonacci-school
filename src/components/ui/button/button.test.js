import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Тест рендера кнопки:", () => {
  it("Кнопка с текстом рендерится без ошибок", () => {
    const tree = renderer.create(<Button text="Кнопка"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка без текста рендерится без ошибок", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Заблокированная кнопка рендерится без ошибок", () => {
    const tree = renderer
      .create(<Button text="Кнопка" disabled={true}></Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Кнопка с индикацией загрузки рендерится без ошибок", () => {
    const tree = renderer
      .create(<Button text="Кнопка" isLoader={true}></Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Нажатие на кнопку вызывает корректный alert", () => {
    window.alert = jest.fn();

    const buttonTextOnClick = "Кнопка работает";

    // Рендерим компонент
    render(
      <Button text="Кнопка" onClick={() => alert(buttonTextOnClick)}></Button>
    );

    // Находим элемент кнопки
    const button = screen.getByRole("button");

    // Имитируем нажатие на ссылку
    fireEvent.click(button);

    // Проверяем, что alert сработал с правильным текстом
    expect(window.alert).toHaveBeenCalledWith(buttonTextOnClick);
  });
});