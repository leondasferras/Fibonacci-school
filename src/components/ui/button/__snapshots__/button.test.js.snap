// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест рендера кнопки: Заблокированная кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button
  
  
  "
  disabled={true}
  type="button"
>
  <p
    className="text "
  >
    Кнопка
  </p>
</button>
`;

exports[`Тест рендера кнопки: Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button
  
  
  "
  type="button"
>
  <p
    className="text "
  />
</button>
`;

exports[`Тест рендера кнопки: Кнопка с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button
  
  loader
  "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тест рендера кнопки: Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button
  
  
  "
  type="button"
>
  <p
    className="text "
  >
    Кнопка
  </p>
</button>
`;
